server/asset/eth/coiner.go

```diff
@@ -197,78 +186,11 @@ func (be *AssetBackend) baseCoin(coinID []byte, contractData []byte) (*baseCoin,
 // and the same account and nonce, effectively voiding the transaction we
 // expected to be mined.
 func (c *swapCoin) Confirmations(ctx context.Context) (int64, error) {
-	swap, err := c.backend.node.swap(ctx, c.backend.assetID, c.secretHash)
-	if err != nil {
-		return -1, err
-	}
-
-	// Uninitiated state is zero confs. It could still be in mempool.
-	// It is important to only trust confirmations according to the
-	// swap contract. Until there are confirmations we cannot be sure
-	// that initiation happened successfully.
-	if swap.State == dexeth.SSNone {
-		// Assume the tx still has a chance of being mined.
-		return 0, nil
-	}
```

Comment:
It wasn't quite clear to me why we wouldn't return `CoinNotFoundError` here. We have the latency queue running for `(Backend).Confirmations` in `(Swapper).processInit`.

Reply:
I don't remember. `CoinNotFoundError` is probably correct.

Reply:
I believe it's because the contract state will not be reflecting txns in txpool.  As such, there is ambiguity here about whether it's really not found or if it is in fact in txpool.  What is the problem with returning `0, nil`?  That would be correct for a transaction not yet mined.